presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -56,31 +54,33 @@ public SerializedPage serialize(Page page)
     {
         SliceOutput serializationBuffer = new DynamicSliceOutput(toIntExact((page.getSizeInBytes() + Integer.BYTES))); // block length is an int
         writeRawPage(page, serializationBuffer, blockEncodingSerde);
+        byte markings = PageCodecMarker.none();
 
         if (!compressor.isPresent()) {
-            return new SerializedPage(serializationBuffer.slice(), UNCOMPRESSED, page.getPositionCount(), serializationBuffer.size());
+            return new SerializedPage(serializationBuffer.slice(), markings, page.getPositionCount(), serializationBuffer.size());
         }
 
         int maxCompressedLength = maxCompressedLength(serializationBuffer.size());
         byte[] compressionBuffer = new byte[maxCompressedLength];
         int actualCompressedLength = compressor.get().compress(serializationBuffer.slice().getBytes(), 0, serializationBuffer.size(), compressionBuffer, 0, maxCompressedLength);
 
         if (((1.0 * actualCompressedLength) / serializationBuffer.size()) > MINIMUM_COMPRESSION_RATIO) {
-            return new SerializedPage(serializationBuffer.slice(), UNCOMPRESSED, page.getPositionCount(), serializationBuffer.size());
+            return new SerializedPage(serializationBuffer.slice(), markings, page.getPositionCount(), serializationBuffer.size());
+        }
+        else {
+            return new SerializedPage(
+                    Slices.copyOf(Slices.wrappedBuffer(compressionBuffer, 0, actualCompressedLength)),
+                    PageCodecMarker.COMPRESSED.mark(markings),
```

Comment:
`COMPRESSED.get()` would be clearer

Reply:
Rather, `PageCodecMarker.of(COMPRESSED)`. Will change.